src/Init/Data/Int/DivMod/Lemmas.lean

```diff
@@ -2894,6 +2894,71 @@ theorem bmod_eq_self_of_le_mul_two {x : Int} {y : Nat} (hle : -y ≤ x * 2) (hlt
+    x / y < x := by
+  by_cases hy' : 1 < y
+  · rw [Int.div_def]
+    unfold Int.ediv
+    obtain ⟨xn, hx'⟩ := Int.eq_ofNat_of_zero_le (a := x) (by omega)
+    obtain ⟨yn, hy'⟩ := Int.eq_ofNat_of_zero_le (a := y) (by omega)
+    simp only [hx', hy', Int.ofNat_eq_coe, Int.ofNat_lt]
+    apply Nat.div_lt_self (by omega) (by omega)
+  · have := @Int.ediv_nonpos_of_nonneg_of_nonpos x y (by omega) (by omega)
+    omega
+
+theorem ediv_nonneg_of_nonneg_of_nonneg {x y : Int} (hx : 0 ≤ x) (hy : 0 ≤ y) :
+    0 ≤ x / y := by
+  rw [Int.div_def]
+  unfold Int.ediv
+  obtain ⟨xn, hx'⟩ := Int.eq_ofNat_of_zero_le (a := x) (by omega)
+  obtain ⟨yn, hy'⟩ := Int.eq_ofNat_of_zero_le (a := y) (by omega)
+  simp only [hx', hy', Int.ofNat_eq_coe, Int.ofNat_lt]
+  exact Int.ofNat_zero_le (xn / yn)
+
+/--  When both x and y are negative we need stricter bounds on x and y
+  to establish the upper bound of x/y, i.e., x / y < x.natAbs.
+  In particular, consider the following counter examples:
+  · (-1) / (-2) = 1 and ¬ 1 < (-1).natAbs
+    (note that Int.neg_one_ediv already handles `ediv` where the numerator is -1)
+  · (-2) / (-1) = 2 and ¬ 1 < (-2).natAbs
+  To exclude these cases, we enforce stricter bounds on the values of x and y.
+-/
+theorem ediv_lt_natAbs_self_of_lt_neg_one_of_lt_neg_one {x y : Int} (hx : x < -1) (hy : y < -1) :
+    x / y < x.natAbs := by
+  rw [Int.div_def]
+  unfold Int.ediv
+  obtain ⟨xn, hx'⟩ := Int.eq_negSucc_of_lt_zero (a := x) (by omega)
+  obtain ⟨yn, hy'⟩ := Int.eq_negSucc_of_lt_zero (a := y) (by omega)
+  simp only [hx', hy', Nat.succ_eq_add_one, Int.ofNat_eq_coe, Int.natCast_add, Int.cast_ofNat_Int]
+  norm_cast
+  rw [natAbs_negSucc, Nat.succ_eq_add_one, Nat.add_lt_add_iff_right,
+    Nat.div_lt_iff_lt_mul (x := xn) (k := yn + 1) (y := xn) (by omega),
+    show (xn < xn * (yn + 1)) = (1 * xn < (yn + 1) * xn) by rw [Nat.one_mul, Nat.mul_comm]]
+  apply Nat.mul_lt_mul_of_lt_of_le (a := 1) (b := xn) (c := yn + 1) (d := xn) (by omega) (by omega) (by omega)
+
+theorem self_le_ediv_of_nonpos_of_nonneg {x y : Int} (hx : x ≤ 0) (hy : 0 ≤ y) :
+    x ≤ x / y := by
+  by_cases hx' : x = 0
+  · simp [hx', zero_ediv]
+  · by_cases hy : y = 0
+    · simp [hy]; omega
+    · simp only [ge_iff_le, Int.le_ediv_iff_mul_le (c := y) (a := x) (b := x) (by omega),
+        show (x * y ≤ x) = (x * y ≤ x * 1) by rw [Int.mul_one], Int.mul_one]
+      apply Int.mul_le_mul_of_nonpos_left (a := x) (b := y) (c  := (1 : Int)) (by omega) (by omega)
+
+theorem neg_self_le_ediv_of_nonneg_of_nonpos (x y : Int) (hx : 0 ≤ x) (hy : y ≤ 0) :
+    -x ≤ x / y := by
+  by_cases hy' : y = 0
+  · simp [hy']; omega
+  · obtain ⟨xn, hx''⟩ := Int.eq_ofNat_of_zero_le (a := x) (by omega)
+    obtain ⟨yn, hy''⟩ := Int.eq_negSucc_of_lt_zero (a := y) (by omega)
+    rw [Int.div_def]
+    unfold Int.ediv
+    simp only [hx'', hy'', Nat.succ_eq_add_one, Int.ofNat_eq_coe, ge_iff_le, Int.neg_le_neg_iff, Int.ofNat_le]
```

Comment:
```suggestion
    simp only [Int.div_def, Int.ediv, hx'', hy'', Nat.succ_eq_add_one, Int.ofNat_eq_coe, ge_iff_le, Int.neg_le_neg_iff, Int.ofNat_le]
```
?

Reply:
this breaks the proof. I can look into it and fix it if necessary, but the proof looked good already imho

Reply:
Let's leave it then. :+1: